baus/subsidies.py

```diff
@@ -36,50 +36,38 @@ def profit_to_prob_func(df):
 
 
 @orca.injectable(cache=True)
-def coffer(policy, scenario):
+def coffer(policy):
     d = {
         "vmt_res_acct":  accounts.Account("vmt_res_acct"),
         "vmt_com_acct":  accounts.Account("vmt_com_acct")
     }
 
-    for key, acct in \
-            policy["acct_settings"]["lump_sum_accounts"].items():
+    for key, acct in policy["acct_settings"]["lump_sum_accounts"].items():
         d[acct["name"]] = accounts.Account(acct["name"])
 
-    for key, acct in \
-            policy["acct_settings"]["office_lump_sum_accounts"].items():
+    for key, acct in policy["acct_settings"]["office_lump_sum_accounts"].items():
         d[acct["name"]] = accounts.Account(acct["name"])    
 
-    if scenario in (policy["acct_settings"]["jobs_housing_fee_settings"]
-                    ["jobs_housing_com_for_res_scenarios"]):
-        for key, acct in \
-                policy["acct_settings"]["jobs_housing_fee_settings"].items():
-            if key != "jobs_housing_com_for_res_scenarios":
-                d[acct["name"]] = accounts.Account(acct["name"])
+    for key, acct in policy["acct_settings"]["jobs_housing_fee_settings"].items():
+        d[acct["name"]] = accounts.Account(acct["name"])
 
     return d
 
 
 @orca.step()
-def preserve_affordable(year, base_year, scenario, policy, residential_units,
-                        taz_geography, buildings, parcels_geography):
-
-    if scenario not in policy["unit_preservation"]["enable_in_scenarios"]:
-        return
+def preserve_affordable(year, base_year, policy, residential_units, taz_geography, buildings, parcels_geography):
 
     # join several geography columns to units table so that we can apply units
     res_units = residential_units.to_frame()
     bldgs = buildings.to_frame()
     parcels_geog = parcels_geography.to_frame()
     taz_geog = taz_geography.to_frame()
 
-    res_units = res_units.merge(bldgs[['parcel_id']], left_on='building_id', 
-                                right_index=True, how='left').\
-        merge(parcels_geog[['gg_id', 'sesit_id', 'tra_id',
-              'juris']], left_on='parcel_id', right_index=True, how='left').\
+    res_units = res_units.merge(bldgs[['parcel_id']], left_on='building_id', right_index=True, how='left').\
```

Comment:
Too much? :) I find stringing merges together to be readable, maybe we could improve the inputs? Not sure how though since this is kind of a logical merging of the units table, parcel geographies, and taz geographies. Maybe all geographies will be on a master table in the future.

Reply:
One of those cases where pep8 makes it harder to read.  Yuqi has this [visually appealing](https://github.com/BayAreaMetro/petrale/blob/cc59eadcb092a083aee9275433f71a1a8c1ea0c0/policies/plu/update_parcels_geography.py#L143) way of chaining them. But it is aesthetics - and a bit of risk of course, that the merge / join will do unexpected things if one of the datasets is missing index keys.